sys/net/crosslayer/ng_pktbuf/ng_pktbuf.c

```diff
@@ -67,18 +67,34 @@ int ng_pktbuf_realloc_data(ng_pktsnip_t *pkt, size_t size)
     return 0;
 }
 
-ng_pktsnip_t *ng_pktbuf_add(ng_pktsnip_t *next, void *data, size_t size,
+ng_pktsnip_t *ng_pktbuf_add(ng_pktsnip_t *payload, void *data, size_t size,
                             ng_nettype_t type)
 {
-    ng_pktsnip_t *snip;
+    ng_pktsnip_t *hdr;
 
     mutex_lock(&_pktbuf_mutex);
 
-    snip = _pktbuf_add_unsafe(next, data, size, type);
+    hdr = _pktbuf_add_unsafe(payload, data, size, type);
 
     mutex_unlock(&_pktbuf_mutex);
 
-    return snip;
+    return hdr;
+}
+
+void ng_pktbuf_hold(ng_pktsnip_t *pkt, unsigned int num)
+{
+    if ((pkt == NULL) || (num == 0)) {
+        return;
+    }
+
+    mutex_lock(&_pktbuf_mutex);
+
+    while (pkt != NULL) {
+        atomic_set_return(&(pkt->users), pkt->users + num);
```

Comment:
actually we do not need to use `atomic_set_return` here, as this sequence is locked by the mutex...